### PR TITLE
Added process method to forecast.Config.

### PR DESCRIPTION
**Pull request by jaheba**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/model/forecast.py [hunk 1]
```diff
@@ -508,42 +511,33 @@ def __repr__(self):
...
 
-    @classmethod
-    def validate(cls, v):
-        if not isinstance(v, str):
-            raise ValueError(
-                f'strict string: expected string value, got {type(v)}'
+        if OutputType.samples in self.output_types:
+            result['samples'] = (
+                forecast.samples.tolist()
+                if hasattr(forecast, 'samples')
```
Comment (lostella) [inline src/gluonts/model/forecast.py]: I thought the `get_samples` method introduced above was meant to avoid this `if`?

Comment (jaheba) [reply src/gluonts/model/forecast.py]: Right. So, do you prefer having a samples method on every Forecast object, or do the logic elsewhere?

Comment (lostella) [reply src/gluonts/model/forecast.py]: I don't have strong preferences, I just wanted to make sure this was intended and the `get_samples` is not some left-over code. The code can go as-is

**Comment by szha** [comment]
Job PR-166/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-166/4/index.html

**Pull request merged**